pkgs/by-name/ls/lspce/package.nix

```diff
@@ -0,0 +1,52 @@
+{
+  lib,
+  callPackage,
+  emacsPackages,
+  fetchFromGitHub,
+  rustPlatform,
+}:
+
+let
+  self = rustPlatform.buildRustPackage {
+    pname = "lspce";
+    version = "1.1.0-unstable-2024-07-13";
+
+    src = fetchFromGitHub {
+      owner = "zbelial";
+      repo = "lspce";
+      rev = "01b77a4f544a912409857083842db51a20bfdbf3";
+      hash = "sha256-oew5EujNYGjk/agBw3ECAVe7GZl8rw/4M5t32JM+1T8=";
+    };
+
+    cargoHash = "sha256-JAeQmeKNwHoakfQchO4xTwP1rg/YWFcdIhWAt8G2FWI=";
+
+    checkFlags = [
+      # flaky test
+      "--skip=msg::tests::serialize_request_with_null_params"
+    ];
+
+    # Move all lib/ files to share/emacs/site-lisp
+    postInstall = ''
+      mkdir -p $out/share/emacs/site-lisp
+      for f in $out/lib/*; do
+        mv $f $out/share/emacs/site-lisp/lspce-module.''${f##*.}
+      done
+      rmdir $out/lib
+    '';
+
+    passthru = {
+      elisp-package = callPackage ./elisp-package.nix {
+        inherit (emacsPackages) f markdown-mode melpaBuild yasnippet;
```

Comment:
This way of introducing elisp dependencies is wrong because dependencies listed here is always from `emacsPackage` which equals to `pkgs.emacs.pkgs`.

For example, if `pkgs.emacs-nox` is used, these dependencies should be from `pkgs.emacs-nox.pkgs`.

Reply:
Oh, how could I forget the elisp packages not being retrieved from Hydra I reported since ~~Heian era~~ https://github.com/NixOS/nixpkgs/issues/324912 ?...